backend/pixeldrain/api_client.go

```diff
@@ -129,6 +130,20 @@ func apiErrorHandler(resp *http.Response) (err error) {
 	return e
 }
 
+func apiRetry(resp *http.Response, err error) (bool, error) {
```

Comment:
This is a neat idea (calling this function with the output of f.src.CallJSON), but I don't think it quite works as the retry function needs to check the context too so I think you'll need to use the standard `shouldRetry` function.

Here is a `shouldRetry` function which should work for you - feel free to adjust the error codes as appropriate. Note that `fserrors.ShouldRetry(err)` is a much more powerful version of `os.IsTimeout` and note also the `fserrors.ContextError(ctx, &err)` which is important as rclone sometimes cancels requests and we don't want to retry them.

```go
// retryErrorCodes is a slice of error codes that we will retry
var retryErrorCodes = []int{
	429, // Too Many Requests.
	500, // Internal Server Error
	502, // Bad Gateway
	503, // Service Unavailable
	504, // Gateway Timeout
}

// shouldRetry returns a boolean as to whether this resp and err
// deserve to be retried.  It returns the err as a convenience
func shouldRetry(ctx context.Context, resp *http.Response, err error) (bool, error) {
	if fserrors.ContextError(ctx, &err) {
		return false, err
	}
	return fserrors.ShouldRetry(err) || fserrors.ShouldRetryHTTP(resp, retryErrorCodes), err
}
```

Reply:
I think think this version will work fine without modification. I've added it in the last commit